bundles/org.eclipse.passage.lic.e4.ui/src/org/eclipse/passage/lic/internal/e4/ui/addons/E4LicensingAddon.java

```diff
@@ -46,10 +43,10 @@ public void applicationStarted(//
 			@SuppressWarnings("unused") //
 			@UIEventTopic(UIEvents.UILifeCycle.APP_STARTUP_COMPLETE) //
 			Event event) {
-		ServiceInvocationResult<ExaminationCertificate> result = new EquinoxPassage()
-				.acquireLicense(new LicensedApplicationFromContext(application).get().identifier());
-		if (new ShouldBeExposed().test(result)) {
-			new WorkbenchNotification(() -> context.get(Shell.class)).expose(result);
+		boolean allowed = new EquinoxPassageUI(() -> context.get(Shell.class))
+				.canUse(new LicensedApplicationFromContext(application).get().identifier());
+		if (!allowed) {
+			new WorkbenchShutdown().run();
```

Comment:
I think the `release` caller should be passed to `WorkbenchShutdown` action as an argument. 
But release will require certificate ...
Hmm, again we need to return a certificate instead of boolean.

Reply:
This is a draft and this guy actually is evolving to `implements Passage` state. In following PRs.